### PR TITLE
PLAT-25917 Catch and handle invalid access token response to upload requests

### PR DESCRIPTION
**Pull request by Fizzadar**

### CODE REVIEW

File: messagix/mercury.go [hunk 1]
```diff
@@ -53,6 +54,9 @@ func (c *Client) SendMercuryUploadRequest(ctx context.Context, threadID int64, m
 
 	_, respBody, err := c.MakeRequest(url, "POST", h, payload, types.NONE)
 	if err != nil {
+		if strings.Contains(err.Error(), "access token is no longer valid") {
+			return nil, ErrTokenInvalidated
+		}
```
Comment (tulir) [inline messagix/mercury.go]: This is just checking if it's the error and returning the same error 🤔

File: messagix/mercury.go [hunk 2]
```diff
@@ -53,6 +54,9 @@ func (c *Client) SendMercuryUploadRequest(ctx context.Context, threadID int64, m
...
 		return nil, fmt.Errorf("failed to send MercuryUploadRequest: %v", err)
```
Comment (tulir) [inline messagix/mercury.go]: this is the actual fix:

```suggestion
		return nil, fmt.Errorf("failed to send MercuryUploadRequest: %w", err)
```

**Review by tulir — CHANGES_REQUESTED** [review]

File: messagix/mercury.go (hunk 1 above)
Comment (tulir) [reply messagix/mercury.go]:
```suggestion
```

Comment (Fizzadar) [reply messagix/mercury.go]: 🤦

**Review by tulir — APPROVED** [review]

File: messagix/http.go [hunk 3]
```diff
@@ -190,7 +190,7 @@ func (c *Client) makeRequestDirect(url string, method string, headers http.Heade
 		}
 	}()
 	if err != nil {
-		c.UpdateProxy(fmt.Sprintf("http request error: %v", err.Error()))
+		c.UpdateProxy(fmt.Sprintf("http request error: %w", err.Error()))
```
Comment (tulir) [inline messagix/http.go]: This one is still wrong:

```suggestion
		c.UpdateProxy(fmt.Sprintf("http request error: %s", err.Error()))
```

**Pull request merged**